src/client/Client.cc

```diff
@@ -8916,6 +8929,21 @@ int Client::statfs(const char *path, struct statvfs *stbuf)
   assert(cct->_conf->client_quota == false || quota_root != nullptr);
 
   if (quota_root && cct->_conf->client_quota_df && quota_root->quota.max_bytes) {
+
+    // Skip the getattr if any sessions are stale, as we don't want to
+    // block `df` if this client has e.g. been evicted, or if the MDS cluster
+    // is unhealthy.
+    if (!_any_stale_sessions()) {
+      int r = _getattr(quota_root, 0, perms, true);
```

Comment:
@jcsp Not sure what to send to `_getattr()` here instead of `perms`? (In jewel the `_getattr` function takes uid and gid, but where do I get those?